### PR TITLE
Simplify settings files

### LINKED ISSUE #7997: simplify our settings files
we have:

- settings.py
- settings_base.py
- dev/settings.py
- stage/settings.py
- prod/settings.py
(and settings_test.py)

with a lot of duplicate settings.  Let's simplify!

### PR DESCRIPTION
**Pull request by eviljeff**
☠️⚠️☠️ May cause breaking of all the things. ☠️⚠️☠️  
fixes #7997 somewhat

### CODE REVIEW

File: src/olympia/lib/settings_base.py [hunk 1]
```diff
@@ -56,8 +56,9 @@ def path(*folders):
 # It puts it in a dir called "workspace".  Way to be, hudson.
 ROOT_PACKAGE = os.path.basename(ROOT)
 
-DEBUG = True
-DEBUG_PROPAGATE_EXCEPTIONS = True
+DEBUG = False
+# Ensure that exceptions aren't re-raised.
+DEBUG_PROPAGATE_EXCEPTIONS = False
```
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: And this one was set to false in all of the settings files.

File: src/olympia/lib/settings_base.py [hunk 2]
```diff
@@ -1616,7 +1617,7 @@ def get_redis_settings(uri):
 
 # If you want to allow self-reviews for add-ons/apps, then enable this.
 # In production we do not want to allow this.
-ALLOW_SELF_REVIEWS = False
+ALLOW_SELF_REVIEWS = True
```
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: I wasn't 100% about this change, but _only_ prod sets it to `False` so it seemed more efficient to default it to `True` in one file than set it to `False` in three.

**Comment by eviljeff** [comment]
Added some fxa config tidy up - dropping internal because it was only used by the internal API we dropped last year.  

I'm dubious having 3 or 2 configs to choose from for each environment is actually required, and certainly doesn't seem wise from a testing/QA point of view (11 variations!) - but they do at least appear to be referenced from addons-frontend.  I think we can take another look when new frontend has edit profile so doesn't need an exception from being redirected to edit profile on account creation.

File: src/olympia/conf/prod/settings.py [hunk 3]
```diff
@@ -161,15 +124,11 @@
...
-
 XSENDFILE_HEADER = 'X-Accel-Redirect'
 
-GEOIP_URL = 'https://geo.services.mozilla.com'
+ALLOW_SELF_REVIEWS = False
 
-AES_KEYS = env.dict('AES_KEYS')
-
-SENTRY_DSN = env('SENTRY_DSN')
+GEOIP_URL = 'https://geo.services.mozilla.com'
```
Comment (EnTeQuAk) [inline src/olympia/conf/prod/settings.py]: I can't find anything where this is being used, neither in our installed dependencies nor in olympia itself. Might be a good candidate to get rid of

File: src/olympia/lib/settings_base.py (hunk 2 above)
Comment (EnTeQuAk) [reply src/olympia/lib/settings_base.py]: Hmm, this is one of the things where I'm unsure as well and would tend to default to the safest value and overwrite it specifically for dev and stage to make things less "safe".

File: src/olympia/conf/prod/settings.py (hunk 3 above)
Comment (eviljeff) [reply src/olympia/conf/prod/settings.py]: There's a quite a few random settings I couldn't find referenced in our code, nor in `settings_base.py`.  A future task might be to reorder them in an import-style - i.e. django settings at the top, then 3rd party, then ours (and sorted alphabetically) to make it slightly easier to work out what's used/needed where.

**Pull request merged**